cmd/crc/cmd/daemon_darwin.go

```diff
@@ -39,13 +44,24 @@ func unixgramListener(vn *virtualnetwork.VirtualNetwork) (*net.UnixConn, error)
 	if err != nil {
 		return conn, errors.Wrap(err, "failed to listen unixgram")
 	}
-	logging.Infof("listening on %s:", constants.UnixgramSocketPath)
+	logging.Infof("listening on %s", constants.UnixgramSocketPath)
 	vfkitConn, err := transport.AcceptVfkit(conn)
 	if err != nil {
 		return conn, errors.Wrap(err, "failed to accept vfkit connection")
 	}
-	err = vn.AcceptVfkit(context.Background(), vfkitConn)
-	return conn, errors.Wrap(err, "failed to accept vfkit connection")
+	if cancel != nil {
+		logging.Warnf("new connection to %s. Closing old connection", constants.UnixgramSocketPath)
+		cancel()
+	}
```

Comment:
To avoid the `cancel`/`ctx` global variables, I’d change `unixgramListener` to have an additional return value of type `CancelFunc`. The cancellation can then be handled in the code calling `vsockListener`.
Even if we keep the code as it is currently, I don’t think `ctx` needs to be global, only `cancel`.

And not related to this change, but I wonder if `unixgramListen` would be a better name for this method instead of `unixgramListener` (but just a comment in passing, nothing important)

Reply:
I split the function into two: `setupUnixgramListener` and `handleUnixgramConnection` since `cancel()` needs to be called once the initial connection is established (`transport.AcceptVfkit` returns). The cancellation is now being handled by the caller function.

For the names, what do you think of the two new functions?

Reply:
`cancel` is only required after the initial connection is established, but I don’t think it’s problematic to call it even if it’s not needed?

Reply:
I think it's needed every time a new connection is established so that it closes the old connection. It is getting stuck in `vn.AcceptVfkit` implying cancellation will help in cleaning up the old goroutine.
Thank you! 😄

Reply:
I was only (badly) paraphrasing your comment "since cancel() needs to be called once the initial connection is established (transport.AcceptVfkit returns)", but yes, it needs to be called every time.

Reply:
oh okay, by my comment I mean every time a new connection is established. Sorry about the ambiguous language 😓